plugin-management-cli/src/main/java/io/jenkins/tools/pluginmanager/cli/CliOptions.java

```diff
@@ -283,7 +286,27 @@ private List<Plugin> getPlugins() {
                 throw new PluginInputException("Unknown file type, file must have .yaml/.yml or .txt extension");
             }
         }
-        return requestedPlugins;
+         List<String> pluginNames = new ArrayList<>();
+
+         File dr = new File(String.valueOf(pluginDir));
+
+         File[] directoryListing = dr.listFiles();
+            if (directoryListing != null) {
+                for (File child : directoryListing) {
+                    pluginNames.add(child.getName());
```

Comment:
This will add many unnecessary items to the list when run in a Jenkins plugins folder that is used by a running Jenkins controller.  I think it would be better to only add the child if it is a plain file (not a directory) and does not have one of the suffixes (".bak", ".pinned", or ".version_from_image").
```suggestion
                    pluginNames.add(child.getName());
```

Reply:
Done @MarkEWaite